src/Esprima/Utils/AstVisitorEventSource.JSX.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using Esprima.Ast;
+
+namespace Esprima.Utils
+{
+    public partial class AstVisitorEventSource
```

Comment:
here again I'm thinking of inheritance, adding needed things to `JsxVisitorEventSource : AstVisitorEventSource`